2015/presentations/sztuczki_z_metoda_super-Dominik-Socek-Dlugajczyk.md

```diff
@@ -0,0 +1,64 @@
+# Temat: Sztuczki z metodą super()
+# Autor: Dominik "Socek" Długajczyk
+Jest jedna funkcjonalność Pythona, której wszyscy używają bardzo często, acz szczątkowo, gdyż panuje opinia iż nadużywanie jej może doprowadzić do nieokreślonego działania aplikacji. Tą funkcjonalnością jest wielodziedziczenie. Postaram się w kilku słowach pokazać jak działa "Method Resolution Order" w Pythonie (wybaczcie, nie znalazłem dobrego tłumaczenia tej nazwy). Postaram się wam pokazać jak można użyć "super()", aby zamockować i przetestować przykładową klasę.
+
+Zacznijmy jednak od teorii. W programowaniu panuje przekonanie, że wielodziedziczenie jest złe, gdyż może występować "the diamond problem", czyli sytuacja, w której klasa dziedziczy po dwóch klasach, a te dwie klasy dziedziczą po jednej i tej samej. Gdybyśmy w takiej sytuacji (powiedzmy w C++) we wszystkich klasach zaimplementowali taką samą metodę, to język nie wiedziałby w jakiej kolejności wykonać dane metody.  W Pythonie ten problem nie występuje dzięki MRO, który to jest implementacją algorytmu "C3 linearization"*. Wprowadzono go w wersji 2.3 (new style classes) oraz w językach Perl 5 i Parrot. Niestety nie są znane mi inne języki, które by implementowały ten algorytm.
```

Comment:
Wydaje mi się, że to zdanie może być trochę mylące:

* W Pythonie ten problem nie występuje dzięki MRO, który to jest implementacją algorytmu "C3 linearization".*

To sugeruje, że MRO w ogóle jako mechanizm w językach programowania jest implementacją C3. Proponowałbym ująć to w trochę inny sposób, na przykład:

*W Pythonie ten problem nie występuje dzięki algorytmowi "C3 linearization", który został wykorzystany do implementacji MRO.*